.github/workflows/main.yml

```diff
@@ -0,0 +1,70 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI Build
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on: [push, pull_request]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  
+  linux:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+    - name: Download latest NuGet (cross-platform)
+      uses: olegtarasov/download-nuget@v1
+
+    - name: Install Dependencies
+      run: |
+         mono $NUGET_EXE restore
+    
+    # Runs a set of commands using the runners shell
+    - name: Build Project
+      run: xbuild TLSharp.Core/TLSharp.Core.csproj
+  macos:
+    # The type of runner that the job will run on
+    runs-on: macos-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@v2
+    - name: Download latest NuGet (cross-platform)
+      uses: olegtarasov/download-nuget@v1
+
+    - name: Install Dependencies
+      run: |
+         mono $NUGET_EXE restore
+    
+    # Runs a set of commands using the runners shell
+    - name: Build Project
+      run: MSBuild TLSharp.Core/TLSharp.Core.csproj
```

Comment:
msbuild should be able to build the .sln file? (not sure about xbuild, but let's try too)

Reply:
xbuild can build .sln but it fails when trying to build Test.VS

Reply:
then let's use xbuild TLSharp.Core.csproj in Linux and msbuild TLSharp.sln in the others

Reply:
https://github.com/aarani/TLSharp/runs/551704535?check_suite_focus=true

Reply:
macos fails to compile VS Unit Tests too. It's because of the mono not xbuild.
https://github.com/aarani/TLSharp/runs/554069446?check_suite_focus=true

Reply:
ok then let's just compile the .sln on Windows

Reply:
Done